### ISSUE TITLE
Suggestions: console home and clear to end of screen + Column suggestions

### ISSUE THREAD

**Issue by jderagon** [willmcgugan/rich#150]
Hello, I am new to python and found this library for a console based program and it's been amazing to work with! The library has been incredibly easy to get up and running with great output and not much overhead. I had looked between blessed and some of the python TUI frameworks, and rich sits at a perfect spot between the two!  I was able to very quickly get some nice output going without much overhead at all and it's been great.

The documentation and examples have been great at helping me work through problems, but it's possible I missed this feature. I'm basically doing a menu for a user interface, so I've been using a combination of blessed and rich. I use the input from blessed, and have some functions to make the rich renderables when it needs to redraw the screen.

The issue is that you cannot clear between redraws because the screen will flash, so you have to draw over top of the existing items by moving home and redrawing. If elements get shorter, you are left with lines that have old output, so you need to clear to the end of the screen as well.

So right now my draw function is basically:
print.(blessed_term.home )
rich_console.print( ui_element )
rich_console.print( ui_element_2 )
...
print( blessed_term.clear_eos )

I think it would be a nice addition if it were possible to add some of these functions to rich, but I understand if they don't fit the design or are not easy to put in there. 

Another suggestion is added features for the column class. The first would be to change the background color of the space between renderables in the column. 

The screen layout I have is, top to bottom, an expanded panel, an expanded table, then a column of several tables, then an expanded panel. The column had a lot of space to the right that that was the color of the terminal background. If I expanded the column, the gaps were still the terminal background color.

What I ended up doing was manually calculating the width of the tables (table borders + padding + width of longest string element), then creating a final panel with box minimal that was the size of the left over space (term width - tables width) that shared a background color with other elements, then added 0 padding, and I had a nice solid background color change for all elements.

It would be nice to be able to just set the background color of the space between the renderables in the column class.

In addition, it would be nice to justify the column elements to the left/right/center/justified positions. Left stacking them to the left, right to the right, center smashing them together in the center, and justify doing what it does now when expand=true. 

Again it's been wonderful working with rich and thank you again for writing it along with great documentation. One last suggestion is that I wanted to have a progress bar inside of a table and thought I couldn't do it until I saw the get_renderable function in the reference, which worked out wonderfully. Might be worth a mention on the progress page as I initially thought I could only use it at the bottom of the screen doing its thing instead of putting it up inside the elements.

tldr:
- console.move_home  console.clear_eos functions
-Column background color
-Column justify for elements left,right,center,justify
-Mention progress.get_renderable() out in main documentation.

**Comment by willmcgugan** [comment]
You can wrap any renderable in padding which will extend padding to the full width.

```python
from rich.padding import Padding
console.print(Padding(ui_element))
```

> console.move_home console.clear_eos functions

Rich may or may not grow such things in the future. Possible as a separate package. Currently the line of separation is that Rich is for rendering only and not interactive. I broke that rule with the Progress display, so we'll see.

> -Column background color

You can set a style when you print something, which effectively does this. Try the following:

```python
from rich.columns import Columns
from rich.console import Console

console = Console()

from rich.table import Table

table = Table("foo", "bar", style="on blue")
table.add_row("1", "2")

console.print(Columns([table]*5, expand=True), style="on green")
```

Or you can wrap it in `Padding(columns, style="on magenta")`, which has a similar effect.

> Column justify for elements left,right,center,justify

I had thought the Align class does this, but its not working as intended. I should have a fix for this in the next version. Essentially you can wrap each column renderable in `Align.center(renderable)`. I'm not certain about making it a per-Columns attribute, but I'll think about it.

> -Mention progress.get_renderable() out in main documentation.

It's `get_renderables` which is recommended for this, and it's mentioned [here](https://rich.readthedocs.io/en/latest/progress.html#customizing). It doesn't go in to detail. Documentation PRs are welcome if you want to expand.

If you have any follow up questions, could you please open an issue per question. Makes it easier to track.

Not sure I fully grasped what you are trying to display with Columns, so perhaps add a screenshot?

**Comment by willmcgugan** [comment]
You may want to try v3.2.0. I've added a `align` parameter to Columns constructor.

**Comment by jderagon** [comment]
Sorry it took me so long to get back to you! Thank you for the help! It looks like wrapping the column in padding did the job exactly as needed! It leaves the tables pushed against each other to the left while clearing the space to the right with the background color needed. 

For what I meant by aligning columns, if you look at what the example code you put up does.
Align left would be to put your hand to the right side of the screen and push all of the boxes to the left so they are against each other and the left side of the screen.
Align right would be put your hand on the left side and push all of the boxes to the right of the screen.
Align center would be put a hand on either side and push them all together.

So it looks like everything was covered here and you've got the alignment going on already in columns, thank you again for the help and update!

**Comment by willmcgugan** [comment]
> For what I meant by aligning columns, if you look at what the example code you put up does.
> Align left would be to put your hand to the right side of the screen and push all of the boxes to the left so they are against > each other and the left side of the screen.
>Align right would be put your hand on the left side and push all of the boxes to the right of the screen.
>Align center would be put a hand on either side and push them all together.

If I understand you correctly, then this would do it:

```python
print(Align.left(columns))
print(Align.right(columns))
print(Align.center(columns))
```

You would need `expand=False` on the Column.
